src/commands.cc

```diff
@@ -1485,6 +1485,20 @@ const CommandDesc debug_cmd = {
             write_to_debug_buffer(format(" * {}:\n{}",
                                   parser[1], dump_regex(compile_regex(parser[1], RegexCompileFlags::Optimize))));
         }
+        else if (parser[0] == "registers")
+        {
+            write_to_debug_buffer("Register info:");
+            for (auto&& [name, reg] : RegisterManager::instance())
+            {
+                auto content = reg->get(context);
+
+                if (content.size() == 1 and content[0] == "")
+                    continue;
+
+                write_to_debug_buffer(format(" * {} = {}\n", name,
+                    join(content | transform([](StringView s) { return format("'{}'", s); }), "\n     = ")));
```

Comment:
there is a function called `quote` in string_utils.hh it will also double up contained `'` to ensure the string matches what Kakoune can parse in command line.

Reply:
Oh, I see now. I was looking at the visual representation, but indeed `echo ''''` just prints a single `'`. Will change that back to your suggestion.